livekit-agents/livekit/agents/llm/_provider_format/anthropic.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import annotations
+
+import base64
+import json
+from dataclasses import dataclass
+from typing import Any
+
+from livekit.agents import llm
+
+
+@dataclass
+class AnthropicFormatData:
+    system_messages: list[str] | None
+
+
+def to_chat_ctx(
+    chat_ctx: llm.ChatContext, *, generating_reply: bool, cache_control: dict[str, Any] | None
+) -> tuple[list[dict], AnthropicFormatData]:
+    messages: list[dict[str, Any]] = []
+    system_messages: list[str] = []
+    current_role: str | None = None
+    content: list[dict[str, Any]] = []
+
+    for i, msg in enumerate(chat_ctx.items):
+        if msg.type == "message" and msg.role == "system":
+            system_messages.append(msg.text_content)
+            continue
+
+        cache_ctrl_i = cache_control if (i == len(chat_ctx.items) - 1) else None
+        if msg.type == "message":
+            role = "assistant" if msg.role == "assistant" else "user"
+        elif msg.type == "function_call":
+            role = "assistant"
+        elif msg.type == "function_call_output":
+            role = "user"
+
+        if role != current_role:
+            if current_role is not None and content:
+                messages.append({"role": current_role, "content": content})
+            content = []
+            current_role = role
+
+        if msg.type == "message":
+            for c in msg.content:
+                if c and isinstance(c, str):
+                    content.append({"text": c, "type": "text", "cache_control": cache_ctrl_i})
+                elif isinstance(c, llm.ImageContent):
+                    content.append(_to_image_content(c, cache_ctrl=cache_ctrl_i))
+        elif msg.type == "function_call":
+            content.append(
+                {
+                    "id": msg.call_id,
+                    "type": "tool_use",
+                    "name": msg.name,
+                    "input": json.loads(msg.arguments or "{}"),
+                    "cache_control": cache_ctrl_i,
+                }
+            )
+        elif msg.type == "function_call_output":
+            content.append(
```

Comment:
Anthropic supports "errored" tool output, I think this PR is a great opportunity to support it. Our OutputItem already has a is_error bool